public/app/ui/components/search/basic/BasicSearchComponent.js

```diff
@@ -0,0 +1,111 @@
+(function () {
+    'use strict';
+
+    define(
+        [
+            'lodash',
+            'knockout'
+        ],
+        function (_, ko) {
+            return function (parameters) {
```

Comment:
This is the view model for the basic search, i.e. the text field and set of buttons to select which field to search on.